main.nf

```diff
@@ -782,6 +782,26 @@ process qualimap {
     """
 }
 
+/*
+ * STEP 9 - preseq
+ */
+process preseq {
+    tag "$name"
+    publishDir "${params.outdir}/preseq", mode: 'copy'
+
+    input:
+    set val(name), file(bam) from ch_bam_for_preseq
+
+    output:
+    file "${bam.baseName}.ccurve.txt" into preseq_results
+
+    script:
+    """
+    samtools sort $bam -o ${bam.baseName}.sorted.bam
```

Comment:
Might want to look into the parallel options for samtools sort:
`  -m INT     Set maximum memory per thread; suffix K/M/G recognized [768M]`
`  -@, --threads INT
               Number of additional threads to use [0]`
My experience is that it can speed things up quite a bit if you have large bam files. But maybe the number of samples is usually filling up the compute anyway?

Reply:
Yes, good point 👍 Same probably goes for the other samtools commands in the pipeline.

Reply:
Looks like I'm already doing all of that in the other `samtools sort` command in the pipeline:
https://github.com/nf-core/methylseq/blob/1f0eaa6f1a8342e98c87b639553f54c47536f9c3/main.nf#L670-L674

And yes, I know it's a waste to do it twice. I started using this one process for both sides of the pipeline but it got really complicated quite quickly. The majority of users use the default bismark pipeline so I figure it's not worth the effort.